openmc/mesh.py

```diff
@@ -1583,6 +1595,42 @@ def indices(self):
                 for t in range(1, nt + 1)
                 for r in range(1, nr + 1))
 
+    @property
+    def lower_left(self):
+        return (self.origin[0]-self.r_grid[-1], self.origin[1]-self.r_grid[-1], self.origin[2]-self.r_grid[-1])
+
+    @lower_left.setter
+    def lower_left(self, lower_left):
+        cv.check_type('mesh lower_left', lower_left, Iterable, Real)
+        cv.check_length('mesh lower_left', lower_left, 1, 3)
+        self._lower_left = lower_left
+
+        if self.upper_right is not None and any(np.isclose(self.upper_right, lower_left)):
+            raise ValueError("Mesh cannot have zero thickness in any dimension")
+
+    @property
+    def upper_right(self):
+        return (self.origin[0]+self.r_grid[-1], self.origin[1]+self.r_grid[-1], self.origin[2]+self.r_grid[-1])
+
+    @upper_right.setter
+    def upper_right(self, upper_right):
+        cv.check_type('mesh upper_right', upper_right, Iterable, Real)
+        cv.check_length('mesh upper_right', upper_right, 1, 3)
+        self._upper_right = upper_right
+
+        if self._width is not None:
+            self._width = None
+            warnings.warn("Unsetting width attribute.")
+
+        if self.lower_left is not None and any(np.isclose(self.lower_left, upper_right)):
+            raise ValueError("Mesh cannot have zero thickness in any dimension")
```

Comment:
Since the lower-left and upper-right corners are calculated based on `origin` and `r_grid`, I think we should just make them read-only properties. That is, get rid of the setters here.

Reply:
@paulromano Hi Dr Romano, so you're saying just delete the DECORATORS? As in, delete @upper_right.setter and @lower_left.setter from lines 1602 and 1615 as shown in the code snippet in the previous comment? And do this for CylindricalMesh and SphericalMesh but.... *not* the RegularMesh?

Reply:
Not just the decorators but the methods under them as well (i.e., lines 1602-1609, 1615-1626). And yes, the reason we have them for `RegularMesh` is because that class is _defined_ by whatever the user specifies for `lower_left` and `upper_right`. In this case, we're just using it for the sake of calculating a bounding box.

Reply:
@paulromano Ok gotcha. I'll take care of it and add a description of the bounding box to docstrings.  

Also, 
I was planning an analogous PR that would add a bounding box property to the RectilinearMesh class. Would such a class require setters for its lower_left and upper_right properties?

Reply:
Nope, for `RectilinearMesh`, the class is "defined" by `x_grid`, `y_grid`, and `z_grid`, so the lower-left and upper-right would be derived properties that don't need setters.